pdns/pdns_recursor.cc

```diff
@@ -4075,6 +4080,12 @@ static int serviceMain(int argc, char*argv[])
     g_log.toConsole(Logger::Critical);
     daemonize();
   }
+  if(Utility::getpid() == 1) {
+    /* We are running as pid 1, register sigterm and sigint handler */
```

Comment:
A mention of why pid 1 is special would be nice. On "normal" systems pid 1 mean we're the init process, so testing on 1 might confuse the reader.

Reply:
See additional code comments